src/cmd/add.rs

```diff
@@ -80,12 +80,28 @@ impl Cmd {
             return Ok(());
         }
 
+        let parent_path = path
+            .parent()
+            .ok_or_else(|| {
+                anyhow!(
+                    "Failed to determine parent path for the repository's new location: {}",
+                    path.to_string_lossy()
+                )
+            })?
+            .to_path_buf();
+
+        if !parent_path.as_path().is_dir() {
+            create_dir_all(&parent_path)?
+        }
```

Comment:
This is not an atomic operation; it can be occur that another process create the directory after checking the path is not a directory, before creating a directory. Can we call only `create_dir_all` [^1] and ignore if `ErrorKind::AlreadyExists` [^2] error occurred while creating?

[^1]: https://doc.rust-lang.org/std/fs/fn.create_dir_all.html
[^2]: https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.AlreadyExists

Reply:
@siketyan 
Thanks for the review! I realized `create_dir_all` doesn't error if the directory already exists. So, I've simplified the code to just call `create_dir_all` without additional checks or handling.

(Though the documentation mentioned it, it wasn't clear enough for me to be confident. However, tracing the process led me to [this code](https://doc.rust-lang.org/src/std/fs.rs.html#2510) which seems to be ignoring errors.)